sys/kern/subr_vmem.c

```diff
@@ -1471,6 +1595,16 @@ vmem_free(vmem_t *vm, vmem_addr_t addr, vmem_size_t size)
 {
 	qcache_t *qc;
 	MPASS(size > 0);
+#ifdef __CHERI_PURE_CAPABILITY__
+	if (vm->vm_flags & VMEM_CAPABILITY_ARENA) {
+		/*
+		 * XXX-AM: These should be checks, not just assertions.
+		 * What do we do on failure? panic?
```

Comment:
I'd say so?

Reply:
Inserted panics regardless of INVARIANTS